Feature/asset list tokens tab

svojsu opened the pull request:
- Introduce new grouping logic for asset groups in `AssetListBaseBuilder`.
- Refactor `AssetListAssetViewModelFactory` to support the new groups.
- Introduce `AssetListCollectionManager` to separate collection view management from controller logic.
- Introduce `AssetListNetworksFlowLayout` and `AssetListTokensFlowLayout`.
- Temporarily disable search logic.
- Temporarily use the filter button to swap layouts.

<img src="https://github.com/user-attachments/assets/d9740e51-3e9a-4e7a-9334-76ed9c41197e" width=40% height=40%>
<img src="https://github.com/user-attachments/assets/a65d1b3e-e174-4922-85f3-fd2dc0de4c79" width=40% height=40%>
<img src="https://github.com/user-attachments/assets/2611edb4-4927-494f-805e-db1f1855c121" width=40% height=40%>
<img src="https://github.com/user-attachments/assets/3b6fe427-e63e-4c14-b7c8-dbf478c2b075" width=40% height=40%>

ERussel reviewed (commented):
1) I think we need to get rid of changes for assets and chains for now as current implementation significantly complicates the logic. So, I would just rebuild all the groups on chain update/insert/delete. On balance update we can still update only those assets for which balance got updated.

2) 1 also fixes the issue with assets enabled/disabled since we rebuild the groups

3) Also noticed that we don't apply hide zero balances to the group by asset in the presenter

svojsu commented:
> 1. I think we need to get rid of changes for assets and chains for now as current implementation significantly complicates the logic. So, I would just rebuild all the groups on chain update/insert/delete. On balance update we can still update only those assets for which balance got updated.
> 2. 1 also fixes the issue with assets enabled/disabled since we rebuild the groups
> 3. Also noticed that we don't apply hide zero balances to the group by asset in the presenter

Builder:

1. Removed the mapping of changes, now we rebuild the groups with lists. Left the changes in balances.
2. Using `MultichainAsset` instead of `ChainAsset` for asset groups.
3. Removed calculators from the model, added them yesterday, and removed the group diffs from the model today.
4. Cleared all unnecessary things from the helpers that were needed for mapping changes.

Presenter:

1. Handled the filter with `hidesZeroBalances`.
2. In the view model, I use balances from the group, not the sum of asset balances.

svojsu commented on novawallet/Modules/AssetList/Base/AssetListAssetsViewModelFactory.swift:
```diff
@@ -35,17 +52,45 @@ class AssetListAssetViewModelFactory {
         self.currencyManager = currencyManager
     }
 
+    func createBalanceViewModel(
```
Actually, we need this method, but reimplemented for groups so we don't calculate the sum again.

svojsu commented on novawallet/Modules/AssetList/Models/AssetListBuilderResult.swift:
```diff
@@ -1,10 +1,30 @@
 import BigInt
+import Operation_iOS
+
+extension ListDifferenceCalculator where T == AssetListChainGroupModel {
```
We don't need it anymore.

ERussel commented on novawallet/Modules/AssetList/View/AssetListCollectionManager/AssetListCollectionViewDataSource.swift:
```diff
@@ -0,0 +1,397 @@
+    func provideAssetCell(
+        _ collectionView: UICollectionView,
+        indexPath: IndexPath,
+        assetIndex _: Int
```
Not used assetIndex
